src/overlays/s07a/Enemy/merylthink.c

```diff
@@ -1189,8 +1189,55 @@ int s07a_meryl_unk_800DCD50( void )
     return 0;
 }
 
-#pragma INCLUDE_ASM("asm/overlays/s07a/s07a_meryl_unk_800DCD58.s")
-#pragma INCLUDE_ASM("asm/overlays/s07a/s07a_meryl_unk_800DCDC8.s")
+int s07a_meryl_unk_800DCD58(WatcherWork *work) {
+    
+    short temp_v0;
+
+    if (!(work->count3 & 0x1F)) {
+        
+        work->field_C04 = -1;
+        work->field_BF0 = -1;
+        s07a_meryl_unk_800DC5B0(work);
+    }
+    
+    temp_v0 = s07a_meryl_unk_800DC5B0(work);
+    work->pad.dir = temp_v0;
```

Comment:
I think these would be `work->pad.dir < 0`

Reply:
You're right, fixed it!